app/services/ExactTargetService.scala

```diff
@@ -151,27 +152,32 @@ class ExactTargetService(
 
     val buildDataExtensionRow =
       GetSalesforceContactForSub(subscription)(zuoraService, salesforceService.repo, defaultContext).map { salesforceContact =>
-        HolidaySuspensionBillingScheduleDataExtensionRow(
-          email = StringOps.oempty(salesforceContact.email).headOption,
-          saltuation = constructSalutation(salesforceContact.title, salesforceContact.firstName, Some(salesforceContact.lastName)),
-          subscriptionName = subscription.name.get,
-          subscriptionCurrency = subscription.plan.charges.price.currencies.head,
-          packageName = packageName,
-          billingSchedule = billingSchedule,
-          numberOfSuspensionsLinedUp,
-          daysUsed,
-          daysAllowed
-        )
+        salesforceContact.email.toRightDisjunction(s"no email in salesforce for ${subscription.id.get}").map { email =>
+          HolidaySuspensionBillingScheduleDataExtensionRow(
+            email = StringOps.oempty(email).headOption,
+            saltuation = constructSalutation(salesforceContact.title, salesforceContact.firstName, Some(salesforceContact.lastName)),
+            subscriptionName = subscription.name.get,
+            subscriptionCurrency = subscription.plan.charges.price.currencies.head,
+            packageName = packageName,
+            billingSchedule = billingSchedule,
+            numberOfSuspensionsLinedUp,
+            daysUsed,
+            daysAllowed
```

Comment:
another thing not introduced in this PR but I would stick to by name params for all of them